src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -61,21 +61,28 @@ public function load($path, ...$paths)
     /**
      * Sets values as environment variables (via putenv, $_ENV, and $_SERVER).
      *
-     * Note that existing environment variables are never overridden.
-     *
      * @param array $values An array of env variables
      */
     public function populate($values)
     {
+        $loadedVars = getenv('SYMFONY_DOTENV_VARS') ? explode(',', getenv('SYMFONY_DOTENV_VARS')) : array();
```

Comment:
`array_flip(explode(',', getenv('SYMFONY_DOTENV_VARS')))` (no need for the check, the resulting array is good enough)